subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/configuration/BuildProcessTest.groovy

```diff
@@ -47,7 +49,46 @@ public class BuildProcessTest extends Specification {
         !buildProcess.configureForBuild(buildParameters(Stub(JavaInfo)))
     }
 
-    def "can only run build when no immutable jvm arguments specified"() {
+    def "all immutable jvm arguments or all immutable system properties need to match"() {
```

Comment:
⭕️ I think it would be better to split this into individual cases. There already is one at the bottom of this class, testing memory settings.